src/Processors/Formats/IOutputFormat.h

```diff
@@ -39,12 +39,15 @@ class IOutputFormat : public IProcessor
 
     RowsBeforeLimitCounterPtr rows_before_limit_counter;
 
+    friend class ParallelFormattingOutputFormat;
```

Comment:
Seems like it's not required

Reply:
We call `consume` inside and it is a protected member of `IOutputFormat`